### ISSUE TITLE
OServerCommandGetGephi: Database instance is not set in current thread. >> HTTP 500

### ISSUE THREAD

**Issue by hhund** [orientechnologies/orientdb#4829]
Hi,

I'm getting a HTTP 500 while running a SQL query against the gephi GET command in orientdb community edition version 2.1.0. Java 1.8.0_51 x64 on Windows 7.

Example Query:
```
http://localhost:2480/gephi/GratefulDeadConcerts/sql/traverse%20V%20from%20written_by/20
````

Http Output:
```javascript
{
  "errors": [{
      "code": 500,
      "reason": 500,
      "content": "com.orientechnologies.orient.core.exception.ODatabaseException: Database instance is not set in current thread. Assure to set it with: ODatabaseRecordThreadLocal.INSTANCE.set(db);"
      }
  ]
}
```

Log Output:
```
2015-08-20 13:33:20:487 FINE  [ONetworkProtocolHttpAbstract.execute] Requested: GET /gephi/GratefulDeadConcerts/sql/traverse%20V%20from%20written_by/20 [ONetworkProtocolHttpDb]
2015-08-20 13:33:20:488 FINE  {db=GratefulDeadConcerts} Read record #5:0 v.1 size=119 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:488 FINE  {db=GratefulDeadConcerts} Read record #4:0 v.2 size=69 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:0 v.2 size=34 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:1 v.1 size=34 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:2 v.1 size=34 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:3 v.1 size=34 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:4 v.1 size=34 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:5 v.2 size=34 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:6 v.2 size=34 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:7 v.2 size=34 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:8 v.2 size=35 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:9 v.2 size=35 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:10 v.2 size=35 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:11 v.1 size=35 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:12 v.2 size=35 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:13 v.2 size=35 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:489 FINE  {db=GratefulDeadConcerts} Read record #12:14 v.2 size=35 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:490 FINE  {db=GratefulDeadConcerts} Read record #12:15 v.2 size=35 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:490 FINE  {db=GratefulDeadConcerts} Read record #12:16 v.2 size=35 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:490 FINE  {db=GratefulDeadConcerts} Read record #12:17 v.2 size=36 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:490 FINE  {db=GratefulDeadConcerts} Read record #12:18 v.1 size=36 bytes [OLocalPaginatedStorage]
2015-08-20 13:33:20:490 FINE  {db=GratefulDeadConcerts} Read record #12:19 v.1 size=35 bytes [OLocalPaginatedStorage]Caught exception
com.orientechnologies.orient.core.exception.ODatabaseException: Database instance is not set in current thread. Assure to set it with: ODatabaseRecordThreadLocal.INSTANCE.set(db);
	at com.orientechnologies.orient.core.db.ODatabaseRecordThreadLocal.get(ODatabaseRecordThreadLocal.java:51)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.checkIfActive(ODatabaseDocumentTx.java:3131)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.close(ODatabaseDocumentTx.java:1129)
	at com.orientechnologies.orient.graph.server.command.OServerCommandGetGephi.execute(OServerCommandGetGephi.java:99)
	at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service(ONetworkProtocolHttpAbstract.java:180)
	at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute(ONetworkProtocolHttpAbstract.java:627)
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:77)

2015-08-20 13:33:20:493 SEVER Internal server error:
com.orientechnologies.orient.core.exception.ODatabaseException: Database instance is not set in current thread. Assure to set it with: ODatabaseRecordThreadLocal.INSTANCE.set(db); [ONetworkProtocolHttpDb]
```

Since I just started trying out orientdb, please advice if this is a bug or a feature.

Regards, Hauke

**Comment by btym** [comment]
Also getting this exception when using the `gephi` command. Mostly happens with large queries (e.q. `TRAVERSE * FROM (SELECT FROM Package WHERE name='lodash') WHILE $depth < 10`) that take several minutes, although it looks like this error happens randomly, it's just more likely that it will happen the longer a query is running. Only affects the `gephi` command, I have not had it happen with the `command` command that the web interface uses.

OrientDB 2.1.0
Ubuntu 14.04.2 LTS
Java 1.7u79

**Comment by lvca** [comment]
I just fixed a similar bug in 2.1.x and develop branches. Could you try please?
